packages/tx/src/eip4844Transaction.ts

```diff
@@ -0,0 +1,479 @@
+   */
+  public getSenderPublicKey(): Buffer {
+    if (!this.isSigned()) {
+      const msg = this._errorMsg('Cannot call this method if transaction is not signed')
+      throw new Error(msg)
+    }
+
+    const msgHash = this.hash()
+    const { v, r, s } = this
+
+    this._validateHighS()
+
+    try {
+      return ecrecover(
+        msgHash,
+        v! + BigInt(27), // Recover the 27 which was stripped from ecsign
+        bigIntToUnpaddedBuffer(r!),
+        bigIntToUnpaddedBuffer(s!)
+      )
+    } catch (e: any) {
+      const msg = this._errorMsg('Invalid Signature')
+      throw new Error(msg)
+    }
+  }
+
+  toJSON(): JsonTx {
+    const accessListJSON = AccessLists.getAccessListJSON(this.accessList)
+    return {
+      chainId: bigIntToHex(this.chainId),
+      nonce: bigIntToHex(this.nonce),
+      maxPriorityFeePerGas: bigIntToHex(this.maxPriorityFeePerGas),
+      maxFeePerGas: bigIntToHex(this.maxFeePerGas),
+      gasLimit: bigIntToHex(this.gasLimit),
+      to: this.to !== undefined ? this.to.toString() : undefined,
+      value: bigIntToHex(this.value),
+      data: '0x' + this.data.toString('hex'),
+      accessList: accessListJSON,
+      v: this.v !== undefined ? bigIntToHex(this.v) : undefined,
+      r: this.r !== undefined ? bigIntToHex(this.r) : undefined,
+      s: this.s !== undefined ? bigIntToHex(this.s) : undefined,
+      maxFeePerDataGas: bigIntToHex(this.maxFeePerDataGas),
+      versionedHashes: this.versionedHashes.map((hash) => bufferToHex(hash)),
+    }
+  }
+
+  _processSignature(v: bigint, r: Buffer, s: Buffer): BlobEIP4844Transaction {
+    const opts = { ...this.txOptions, common: this.common }
+
+    return BlobEIP4844Transaction.fromTxData(
+      {
+        chainId: this.chainId,
+        nonce: this.nonce,
+        maxPriorityFeePerGas: this.maxPriorityFeePerGas,
+        maxFeePerGas: this.maxFeePerGas,
+        gasLimit: this.gasLimit,
+        to: this.to,
+        value: this.value,
+        data: this.data,
+        accessList: this.accessList,
+        v: v - BigInt(27), // This looks extremely hacky: @ethereumjs/util actually adds 27 to the value, the recovery bit is either 0 or 1.
+        r: bufferToBigInt(r),
+        s: bufferToBigInt(s),
+        maxFeePerDataGas: this.maxFeePerDataGas,
+        versionedHashes: this.versionedHashes,
+        blobs: this.blobs,
+        kzgCommitments: this.kzgCommitments,
+      },
+      opts
+    )
+  }
+  /**
+   * Return a compact error string representation of the object
+   */
+  public errorStr() {
+    let errorStr = this._getSharedErrorPostfix()
+    errorStr += ` maxFeePerGas=${this.maxFeePerGas} maxPriorityFeePerGas=${this.maxPriorityFeePerGas}`
+    return errorStr
+  }
+
+  /**
+   * Internal helper function to create an annotated error message
+   *
+   * @param msg Base error message
+   * @hidden
+   */
+  protected _errorMsg(msg: string) {
+    return `${msg} (${this.errorStr()})`
+  }
+
+  public txData() {
```

Comment:
Hmm, I have thought about this method a bit. Very generic naming *and* functionality (from some point of view), but then only introduced here for the 4844 txs for this serialization use case.

Not sure what to make of it actually, but also not totally decided if this combination is well chosen or not. Options might be to  1. do a more specific name here, 2. plan to expand this method to other tx types (not sure if makes sense), 3. keep internal for now, 4. Just keep as implemented.

Just throwing in the ring here, totally no opinion on a direction yet.

Reply:
Totally agree that the name is inconvenient.  `toJSON()` might be more appropriate.  I basically just built it as a way to easily get the `txData` fields I need to use in the transaction constructor since I need to quickly be able to go from minimal to network wrapper and back again without having to manually extract the necessary fields.  There may be a better way to do this (just make it private, maybe?)

Reply:
`toNetworkWrapperFormat()` ? 🤔

Reply:
Hmm, that could work.  At the end of the day, it's really just an internal helper method for constructing the data needed for the `ssz.serialize` call for the Network wrapper.  I'll convert it to a simple helper that's not part of the main class.  There's no reason it has to be part of the class API.

Reply:
Yeah, that's for sure a good way to go for now as well, respectively maybe the more solid choice.